packages/perseus-score/src/widgets/free-response/validate-free-response.ts

```diff
@@ -0,0 +1,20 @@
+import type {
+    PerseusFreeResponseUserInput,
+    ValidationResult,
+} from "@khanacademy/perseus-core";
+
+/**
+ * Checks user input from the free response widget to see if it is scorable.
+ * Since the input is free text, we only check that it is not empty.
+ */
+function validateFreeResponse(
+    userInput: PerseusFreeResponseUserInput,
+): ValidationResult {
+    if (userInput.currentValue.trim() === "") {
+        return {type: "invalid", message: "The answer cannot be empty."};
```

Comment:
We can't use strings here. The problem is that the validation code will run on both the client and the server. Rather than having the server have access to all strings or trying to map `en` strings to other locales, we decided that the server will return error codes and those codes will get mapped to translated strings.

See `validateCategorizer`.